pkg/xray/resource_xray_settings_test.go

```diff
@@ -0,0 +1,50 @@
+package xray
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestDbSyncTime(t *testing.T) {
+	_, fqrn, resourceName := mkNames("db_sync-", "xray_settings")
+	time := "18:45"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviders(),
+		Steps: []resource.TestStep{
+			{
+				Config: dbSyncTime(resourceName, time),
+				Check:  resource.TestCheckResourceAttr(fqrn, "db_sync_updates_time", time),
+			},
+		},
+	})
+}
+
+func TestDbSyncTimeNegative(t *testing.T) {
+	_, _, resourceName := mkNames("db_sync-", "xray_settings")
+	var invalidTime = []string{"24:00", "24:55", "", "12:0", "string", "12pm"}
```

Comment:
`9:00` must also be rejected. I know xray will accept it, but it's not proper 24h time

Reply:
I think we need to accept everything Xray will accept.

Reply:
I've changed regex to reject H:mm. Updated the test